src/libslic3r/PrintObject.cpp

```diff
@@ -428,6 +432,25 @@ void PrintObject::generate_support_material()
     }
 }
 
+void PrintObject::prepare_adaptive_infill_data()
+{
+    float fill_density = this->print()->full_print_config().opt_float("fill_density");
+    float infill_extrusion_width = this->print()->full_print_config().opt_float("infill_extrusion_width");
+
+    coordf_t line_spacing = infill_extrusion_width / ((fill_density / 100.0f) * 0.333333333f);
+
+    BoundingBoxf bed_shape(this->print()->config().bed_shape.values);
+    BoundingBoxf3 printer_volume(Vec3d(bed_shape.min(0), bed_shape.min(1), 0),
+            Vec3d(bed_shape.max(0), bed_shape.max(1), this->print()->config().max_print_height));
+
+    Vec3d model_center = this->model_object()->bounding_box().center();
+    model_center(2) = 0.0f; // Set position in Z axis to 0
+    // Center of the first cube in octree
+
+    TriangleMesh mesh = this->model_object()->mesh();
```

Comment:
ModelObjec::mesh() returns all meshes of all instances, while you want just a mesh representing a single instance, thus you want to call ModelObject::raw_mesh()

Thinking of that, raw_mesh() will NOT re-orient triangle normals for mirrored objects. 

See
std::vector<ExPolygons> PrintObject::slice_volumes(const std::vector<float> &z, SlicingMode mode, const std::vector<const ModelVolume*> &volumes) const

for how the meshes of the volumes are collected and transformed to the PrintObject coordinate space.